### PR TITLE
fix: Instance entity completion

### PR DESCRIPTION
**Pull request by llun**
## Summary

- Complete the v2 Instance entity (`GET /api/v2/instance`): `usage.users.active_month` (reuses the cached nodeinfo active-month query — distinct local actors with a status in the last 30 days), `thumbnail`/`icon` backed by the bundled `public/logo.png` + `icon-192/512.png`, `api_versions: { mastodon: 2 }`, `contact { email, account }`, `configuration.urls.streaming`, `configuration.vapid.public_key` from the push config, and `registrations.enabled` now reflects `config.registrationOpen` instead of being hardcoded `false`.
- Complete the v1 Instance entity (`GET /api/v1/instance`): `stats { user_count, status_count, domain_count }` as integers (per the V1::Instance entity; only `instance/activity` stringifies), `urls.streaming_api`, `contact_account`, `rules` (same source as `/api/v1/instance/rules`), a real contact email instead of the `'-'` placeholder, plus consistency alignment of `registrations` and `thumbnail` with v2.
- New `getInstanceAdminActorId()` database lookup (earliest-created local actor of an `accounts.role = 'admin'` account) feeding the contact account through the existing `getMastodonActorFromId` serializer.
- New shared helpers in `lib/services/mastodon/instance.ts` (`getInstanceContactEmail`, `getInstanceStats`, `getInstanceContactAccount`); all are throw-free so the public instance endpoints degrade to zeros/null instead of 500ing when the database is down.

## Decisions

- Streaming keys are emitted present-but-empty (`''`): no streaming API exists, empty strings are falsy so JS clients fall back to REST polling, and advertising `wss://<domain>` would cause clients to hammer a nonexistent endpoint.
- Contact email is derived (no new env var): `ACTIVITIES_EMAIL_FROM` sender → VAPID contact (`mailto:` stripped) → `''`.
- `api_versions.mastodon = 2` via `MASTODON_INSTANCE_API_VERSION` (the value Mastodon 4.3 introduced the field with; single-constant bump later).

## Notes

- No migrations — the reference schema dumps are unchanged.
- Part of Phase 2 (Entity & Parameter Gaps), PR 2.4. Independent of the other Phase 2 PRs.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: app/api/v2/instance/route.test.ts [hunk 1]
```diff
@@ -123,8 +212,40 @@ describe('GET /api/v2/instance', () => {
...
+      expect(body.contact).toEqual({
+        email: 'admin@llun.test',
+        account: null
+      })
     } finally {
       mockDatabase = database
     }
   })
+
+  it('maps usage.users.active_month from the active-month stat, not another counter', async () => {
```
Comment (llun) [inline app/api/v2/instance/route.test.ts:225]: 🤖 **Sub-agent review (tests dimension), addressed.**

The v2 `usage.users.active_month` assertion was all-zero (the test never seeded data, so every stat is 0), so a wrong-field mapping like `active_month: stats.userCount` would ship green. Fixed in 38f8d7489681ed8e35db7ddcdfa6f9b813a1c5d9: this test stubs the stats source with distinct counters (activeMonth=6 vs userCount=9/statusCount=15/domainCount=4) and asserts active_month is the active-month value specifically.

File: app/api/v1/instance/route.test.ts [hunk 2]
```diff
@@ -45,4 +90,88 @@ describe('GET /api/v1/instance', () => {
...
+        domain_count: 0
+      })
+      expect(body.rules).toEqual([])
+      expect(body.contact_account).toBeNull()
+    } finally {
+      mockDatabase = database
+    }
+  })
+
+  it('degrades to an empty rules list when getInstanceRules rejects but the database is present (still 200)', async () => {
```
Comment (llun) [inline app/api/v1/instance/route.test.ts:157]: 🤖 **Sub-agent review (tests dimension), addressed.**

The v1 rules `try/catch` (degrade to `rules: []` on a query failure, avoiding a 500 on the public endpoint) was untested — the only degradation test set `mockDatabase = null`, short-circuiting the `if (database)` guard before the catch. Deleting the catch would have left all tests green. Fixed in 38f8d7489681ed8e35db7ddcdfa6f9b813a1c5d9: this test uses a present database whose `getInstanceRules()` rejects and asserts 200 + `rules: []`.

File: lib/database/sql/instanceActivity.ts [hunk 3]
```diff
@@ -230,6 +230,22 @@ export const getInstancePeersFromActors = async (
...
+): Promise<string | null> => {
+  // Remote actors have a null accountId, so the inner join naturally limits
+  // the lookup to local actors. Earliest-created wins so the contact account
+  // is stable across requests.
+  const row = await database('actors')
+    .join('accounts', 'actors.accountId', 'accounts.id')
+    .where('accounts.role', 'admin')
+    .whereNull('actors.deletionStatus')
+    .orderBy('actors.createdAt', 'asc')
+    .first<{ id: string } | undefined>('actors.id as id')
```
[the remark this answers is not in the document]
Comment (llun) [reply lib/database/sql/instanceActivity.ts]: Done in 81b251ca9139b93a185a324f274d9c1fbc8cddcf — dropped the redundant `| undefined`; `.first<{ id: string }>()` already yields `{ id: string } | undefined`. Build green.

File: lib/services/mastodon/instance.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  domainCount: number
+  activeMonth: number
+}
+
+export const EMPTY_INSTANCE_STATS: InstanceStats = {
+  userCount: 0,
+  statusCount: 0,
+  domainCount: 0,
+  activeMonth: 0
+}
```
[the remark this answers is not in the document]
Comment (llun) [reply lib/services/mastodon/instance.ts]: Done in 81b251ca9139b93a185a324f274d9c1fbc8cddcf — `EMPTY_INSTANCE_STATS` is now `Object.freeze(...)` typed `Readonly<InstanceStats>`. (`Readonly<InstanceStats>` is still assignable to the `InstanceStats` return type, so `getInstanceStats` compiles unchanged; build + full suite green.)

**Comment by llun** [comment]
### 🤖 Review loop converged (2 rounds) — ready for maintainer review

- **R1** — sub-agent (5 dimensions + adversarial verify): (a) the v2 `usage.users.active_month` assertion was all-zero and couldn't catch a wrong-field mapping; (b) the v1 `getInstanceRules` failure branch (degrade to `rules: []`) was untested (the only degradation test set `mockDatabase=null`, short-circuiting before the catch). gemini: freeze the `EMPTY_INSTANCE_STATS` singleton (returned by reference) and drop a redundant `.first<>` generic. All fixed.
- **R2** — sub-agent: 0 findings; gemini: clean summary, no new threads → **converged**.

All review threads resolved. Local gate green (lint 0, build, 5579 tests); CI running on the final commit.
